### PR TITLE
WriterToParquet partial rewrite to support writing partitioned data.

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: storey/writers.py [hunk 1]
```diff
@@ -19,8 +20,8 @@
 
 class _Writer:
     def __init__(self, columns: Union[str, List[str], None], infer_columns_from_data: Optional[bool],
-                 index_cols: Union[str, List[str], None] = None, retain_dict: bool = False,
-                 storage_options: Optional[dict] = None):
+                 index_cols: Union[str, List[str], None] = None, partition_cols: Union[str, List[str], None] = None,
+                 retain_dict: bool = False, storage_options: Optional[dict] = None):
```
Comment (katyakats) [inline storey/writers.py:24]: please add documentation for these new params

Comment (gtopper) [reply storey/writers.py:24]: There's only one new param :)

Comment (gtopper) [reply storey/writers.py:24]: And it's actually already documented 😄 
Just need to update the documentation.

Comment (gtopper) [reply storey/writers.py:24]: Updated.

**Review by katyakats — APPROVED** [review]

**Pull request merged**